espressif/idf-eclipse-plugin
# IEP-683 Updating esp-idf master directly from plugin

## Description

Please include a summary of the change and which issue is fixed.

Fixes # ([IEP-683](https://jira.espressif.com:8443/browse/IEP-683))

## Type of change

- New feature (non-breaking change which adds functionality)

## How has this been tested?

Case 1:
- Download and select the ESP-IDF master version (Espressif -> Download and Configure ESP-IDF)
- By using a terminal reset the master version to the older commit (`git reset --hard <commit_id>`) and use `git clean -fd` to prevent untracked changes
- Update ESP-IDF from the plugin (Espressif -> ESP-IDF Tools Manager - Update ESP-IDF master)

Case 2:
-  Download and select the ESP-IDF release version (Espressif -> Download and Configure ESP-IDF)
- Check if the Update ESP-IDF command is not visible (Espressif -> ESP-IDF Tools Manager - Update ESP-IDF master)

**Test Configuration**:
* ESP-IDF Version:master
* OS (Windows,Linux and macOS): All

## Dependent components impacted by this PR:

- Espressif menu commands 

## Checklist
